core/src/main/java/org/mskcc/cbio/portal/scripts/ImportExtendedMutationData.java

```diff
@@ -411,35 +417,53 @@ public void importData() throws IOException, DaoException {
-                                                    mutation.setEvent(event);
-                                                } else {
-                                                    mutation.setMutationEventId(++mutationEventId);
-                                                    existingEvents.put(mutation.getEvent(), mutation.getEvent());
-                                                    newEvents.add(mutation.getEvent());
-                                                }
-
-                                                ExtendedMutation exist = mutations.get(mutation);
-                                                if (exist!=null) {
-                                                    ExtendedMutation merged = mergeMutationData(exist, mutation);
-                                                    mutations.put(merged, merged);
-                                                } else {
-                                                    mutations.put(mutation,mutation);
-                                                }
-                                                if(!sampleSet.contains(sample.getStableId())) {
-                                                    addSampleProfileRecord(sample);
-                                                }
-                                                //keep track:
-                                                sampleSet.add(sample.getStableId());
-                                                geneSet.add(mutation.getEntrezGeneId()+"");
+                        MutationEvent event =
+                            existingEvents.containsKey(mutation.getEvent()) ?
+                            existingEvents.get(mutation.getEvent()) :
+                            DaoMutation.getMutationEvent(mutation.getEvent());
+                        if (event!=null) {
+                            mutation.setEvent(event);
+                        } else {
+                            mutation.setMutationEventId(++mutationEventId);
+                            existingEvents.put(mutation.getEvent(), mutation.getEvent());
+                            newEvents.add(mutation.getEvent());
+                        }
+
+                        ExtendedMutation exist = mutations.get(mutation);
+                        if (exist!=null) {
+                            ExtendedMutation merged = mergeMutationData(exist, mutation);
+                            mutations.put(merged, merged);
+                        } else {
+                            mutations.put(mutation,mutation);
+                        }
+                        if(!sampleSet.contains(sample.getStableId())) {
+                            addSampleProfileRecord(sample);
+                        }
+                        // update ascn object with mutation unique key details
+                        if (ascn != null){
+                            ascn.updateAscnUniqueKeyDetails(mutation);
```

Comment:
I can't find this method in the AlleleSpecificCopyNumber class of this PR.  I just checked cgds.sql, do we have    this because we don't have an dbms generated int in the mutation table.  Maybe we should introduce this to make this cleaner.